frontend/src/app/services/git/git.service.ts

```diff
@@ -23,29 +22,28 @@ export interface Instance {
   providedIn: 'root',
 })
 export class GitService {
-  base_url = new URL('git-utils/', environment.backend_url + '/');
+  BACKEND_URL_PREFIX = environment.backend_url + '/projects/';
+  base_url = new URL('projects/', environment.backend_url + '/');
+
+  private _instance = new BehaviorSubject<Instance | undefined>(undefined);
+
+  readonly instance = this._instance.asObservable();
 
   constructor(private http: HttpClient) {}
 
-  instance: Instance | null = null;
-
-  fetch(model_slug: string, credentials: Credentials): Observable<Instance> {
-    let url = new URL('revisions/', this.base_url);
-    return new Observable<Instance>((subscriber) => {
-      this.http
-        .get<Instance>(url.toString(), {
-          params: {
-            model_slug,
-            url: credentials.url,
-            username: credentials.username,
-            password: credentials.password,
-          },
-        })
-        .subscribe((value) => {
-          this.instance = value;
-          subscriber.next(value);
-          subscriber.complete();
-        });
-    });
+  loadInstance(
```

Comment:
The naming is confusing here. It's not really loading a instance, it's fetching the revisions of a git repository.